tools/unit_tests.py

```diff
@@ -3,65 +3,45 @@
 import util
 import sys
 import subprocess
-import re
+import itertools
+import http_server
 
 
-def run_unit_test2(cmd):
+def run_unit_test(cmd):
     process = subprocess.Popen(
         cmd,
         bufsize=1,
         universal_newlines=True,
         stdout=subprocess.PIPE,
         stderr=subprocess.STDOUT)
-    (actual, expected) = util.parse_unit_test_output(process.stdout, True)
+    util.parse_unit_test_output(process.stdout, True)
     process.wait()
     errcode = process.returncode
     if errcode != 0:
         sys.exit(errcode)
-    # To avoid the case where we silently filter out all tests.
-    assert expected > 0
-    if actual == None and expected == None:
-        raise AssertionError("Bad js/unit_test.ts output")
-    if expected != actual:
-        print "expected", expected, "actual", actual
-        raise AssertionError("expected tests did not equal actual")
```

Comment:
This logic should be ported over to the runner too

Reply:
The thing is: up till now when one of tests failed remaining tests for other permissions where not run. I think we can safely spawn subprocesses for each permissions combination and return failure if any of them failed. That way all tests will always run, what do you think?

Reply:
That sounds fine. But for each set of permissions there is a summary line printed. For example:
```
Running tests for: --allow-write --allow-run
running 1 tests
test runWithCwdIsAsync ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```
The js/unit_tests_runner.ts needs to parse the first line (to determine how many tests there are) and then also parse the last line to check that passed matches. If we don't have this feature, it's possible for Deno to silently crash with exit code 0 and the tests all pass.

Reply:
You're right, I'll add it

Reply:
Okay, so I've ported it to TS, but the thing is... There is no line-per-line output of unit tests ATM because there's no available `lines` helper. I didn't want to copy over `chunks` implemented by @kevinkassimo from `xeval.ts` and there's no way to use it, because that file uses extensionless imports.

Reply:
CC @ry

Reply:
Just copy it over add a todo ?

Reply:
Will do

Reply:
Okay I tried... It's not really possible since `chunk` from `xeval.ts` uses import from `js/buffer.ts` so again there's conflict because of extensionless imports. I'm gonna leave it as is for now...